battle-service/src/main/java/org/astashonok/battleservice/rest/BattleController.java

```diff
@@ -0,0 +1,41 @@
+package org.astashonok.battleservice.rest;
+
+import org.astashonok.battleservice.dtos.BattleDto;
+import org.astashonok.battleservice.models.BattleCreationForm;
+import org.astashonok.battleservice.models.BattleState;
+import org.astashonok.battleservice.models.MoveForm;
+import org.astashonok.battleservice.services.BattleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/battles")
+public class BattleController {
+
+    @Autowired
+    private BattleService battleService;
+
+    @PostMapping("/make-move")
+    public BattleState makeMove(@RequestBody @Validated MoveForm moveForm) {
+        return battleService.makeMove(moveForm);
+    }
+
+    @GetMapping("/opened")
+    public List<BattleDto> getOpenedBattles() {
+        return battleService.getOpenedBattles();
+    }
+
+    @PostMapping("/{battleId}/join")
+    public BattleDto joinBattle(@PathVariable UUID battleId) {
+        return battleService.join(battleId);
```

Comment:
try to reproduce situation when multiple users call this endpoint at the same time. i assume it's not warranty order of joiner. 

in case  user 1,2,3 try to join in the same time , there is a chance that user 3 will be only one who joined , but 1 and 2 could receive success response code. 


try to solve this issue.